application/src/main/java/org/togetherjava/tjbot/commands/filesharing/ShareLongAttachmentsMessageListener.java

```diff
@@ -0,0 +1,174 @@
+import net.dv8tion.jda.api.interactions.components.buttons.Button;
+import net.dv8tion.jda.api.interactions.components.buttons.ButtonStyle;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+public class ShareLongAttachmentsMessageListener extends MessageReceiverAdapter {
+
+    private static final ObjectMapper JSON = new ObjectMapper();
+    private static final Logger logger =
+            LoggerFactory.getLogger(ShareLongAttachmentsMessageListener.class);
+
+    private final Set<String> extensions = Set.of("txt", "java", "gradle", "xml", "kt", "json",
+            "fxml", "css", "c", "h", "cpp", "py", "yml");
+
+    private static final String API_KEY = "< placeholder >";
+    private static final String API = "https://api.github.com/gists";
+    private static final HttpClient CLIENT = HttpClient.newHttpClient();
+
+    private final Predicate<String> isStagingChannelName;
+    private final Predicate<String> isOverviewChannelName;
+
+    public ShareLongAttachmentsMessageListener(@NotNull Config config) {
+        super(Pattern.compile(".*"));
+
+        isStagingChannelName = Pattern.compile(config.getHelpSystem().getStagingChannelPattern())
+            .asMatchPredicate();
+        isOverviewChannelName = Pattern.compile(config.getHelpSystem().getOverviewChannelPattern())
+            .asMatchPredicate();
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+
+        if (!isHelpThread(event)) {
+            return;
+        }
+
+        event.getMessage()
+            .getAttachments()
+            .stream()
+            .filter(attachment -> extensions.contains(attachment.getFileExtension()))
+            .forEach(attachment -> processAttachment(event, attachment));
+    }
+
+    private void processAttachment(MessageReceivedEvent event, Message.Attachment attachment) {
+        attachment.retrieveInputStream()
+            .thenApply(this::readAttachment)
+            .thenApply(content -> uploadToSharingService(event, attachment, content))
+            .thenAccept(response -> sendMessage(event, attachment, response));
+    }
+
+    private String readAttachment(@NotNull InputStream stream) {
+        try {
+            byte[] bytes = stream.readAllBytes();
+            return new String(bytes, StandardCharsets.UTF_8);
```

Comment:
NIT: If you want, you can put this into a single line `return new String(stream.readAllbytes(), ...)`. Your choice though. That said, the name `bytes` is bad, `rawData` or `rawContent` would be better. Or even `rawAttachmentContent`.

Variable names should reflect what they logically represent in the context of the code and what they are used for. And not their type. `byte[]` already tells me that its `bytes`. Use the variable name to explain the context instead.